static/js/message_events.js

```diff
@@ -177,6 +185,21 @@ exports.update_messages = function update_messages(events) {
                         }
                     }
                 }
+
+                if (current_filter && stream_changed &&
+                    current_filter.has_topic(stream_name, orig_topic)) {
+                    const new_filter = current_filter.filter_with_new_params({
+                        operator: 'stream',
+                        operand: new_stream_name,
+                    });
+                    const operators = new_filter.operators();
+                    const opts = {
+                        trigger: 'stream change',
```

Comment:
This can be changed to 'message edit` when you combine with the previous block -- this value isn't parsed, it's just for debugging.